pkg/sources/elasticsearch/unit_of_work.go

```diff
@@ -0,0 +1,101 @@
+package elasticsearch
+
+import "fmt"
+
+type DocumentSearch struct {
+	index         *Index
+	offset        int
+	documentCount int
+	skipCount     int
+	filterParams  *FilterParams
+}
+
+type UnitOfWork struct {
+	maxDocumentCount int
+	documentCount    int
+	documentSearches []DocumentSearch
+}
+
+func NewUnitOfWork(maxDocumentCount int) UnitOfWork {
+	uow := UnitOfWork{maxDocumentCount: maxDocumentCount}
+	uow.documentSearches = []DocumentSearch{}
+
+	return uow
+}
+
+func (ds *DocumentSearch) String() string {
+	if ds.offset > 0 {
+		return fmt.Sprintf("%s [%d:]", ds.index.name, ds.offset)
+	} else {
+		return ds.index.name
+	}
+}
+
+func (uow *UnitOfWork) AddSearch(
```

Comment:
**Question**: any reason we need to export this function? Looks like the only place it's being used is `distributeDocumentScans` and the tests.

Reply:
Yeah you're right, no need to export here 👍